persistence/src/main/java/de/cyface/persistence/PersistenceLayer.java

```diff
@@ -565,6 +565,50 @@ public void deletePoint3dData(final long measurementIdentifier) {
         }
     }
 
+    /**
+     * Loads the "cleaned" {@link Track}s for the provided {@link Measurement}.
+     * <p>
+     * This method loads the complete {@code Track}s into memory. For large {@code Track}s this could slow down the
+     * device or even reach the applications memory limit.
+     *
+     * TODO [MOV-554]: provide a custom list implementation that loads only small portions into memory.
+     *
+     * @param measurementIdentifier The id of the {@code Measurement} to load the track for.
+     * @param locationCleaningStrategy The {@link LocationCleaningStrategy} used to filter the
+     *            {@link GeoLocation}s
+     * @return The {@link Track}s associated with the {@code Measurement}. If no {@code GeoLocation}s exists, an empty
+     *         list is returned.
+     * @throws CursorIsNullException when accessing the {@code ContentProvider} failed
+     */
+    @SuppressWarnings("unused") // Used by SDK implementing apps (SR, CY)
+    public List<Track> loadTracks(final long measurementIdentifier,
+            @NonNull final LocationCleaningStrategy locationCleaningStrategy) throws CursorIsNullException {
+
+        final List<Track> rawTracks = loadTracks(measurementIdentifier);
+        final List<Track> cleanedTrack = new ArrayList<>();
+        int numberOfRawPoints = 0;
+        int numberOfCleanPoints = 0;
+        for (final Track rawTrack : rawTracks) {
+            numberOfRawPoints += rawTrack.getGeoLocations().size();
+            final Track track = new Track();
+            // Only add locations considered "clean"
+            for (final GeoLocation location : track.getGeoLocations()) {
+                if (locationCleaningStrategy.isClean(location)) {
+                    track.add(location);
+                    numberOfCleanPoints++;
+                }
+            }
+            if (track.getGeoLocations().size() > 0) {
+                cleanedTrack.add(track);
+            }
+        }
+        Log.d(TAG,
+                String.format("Cleaned track: %d of %d sub tracks and %d of %d locations filtered.",
+                        (rawTracks.size() - cleanedTrack.size()), rawTracks.size(),
+                        (numberOfRawPoints - numberOfCleanPoints), numberOfRawPoints));
+        return cleanedTrack;
+    }
+
```

Comment:
Doing such stuff in Java code is most likely a performance bottleneck. This could be achieved directly in SQL which should be way faster and avoid loading points we do not need into memory on the first hand. We should probably talk about this. I have a bad feeling about this (https://youtu.be/S74rvpc6W60?t=9).

Reply:
You are totally right, I focused only on reusing existing code instead of minimizing resources. I'll add a custom cursor for this database access.

Reply:
Implemented. Please review the two new commits and approve.